### ISSUE TITLE
Handle config building in a decent way

### ISSUE THREAD

**Issue by veruu** [CKI-project/skt#396]
Logs from config making are not added to the buildlog. This should be easily fixable by adding an optional argument to the config preparation methods (for subprocess's stderr/stdout) and moving the `__prepare_kernel_config()` call in `mktgz()` into the block with writer/reader

Having the above fixed, we still need to deal with `getrelease()` which is called after the build failed/finished. The method expects the config files to be ready and calls `__prepare_kernel_config()` again if they are not and "unexpectedly" raises an exception.

One of the options would be to make it clear that `getrelease()` is only to be called after a successful config build (otherwise it's useless for us anyways, since we won't be running any tests with a kernel that doesn't build). It could return `None` if called before or after a failed build instead of calling the config preparation.

@spbnick also suggested it might be useful to call `__prepare_kernel_config()` method explicitly, handle potential issues that arise and only then call the actual build. However, this idea doesn't handle the mess around `getrelease()` nor logging the config making process (and would actually make the logging more complicated). However, maybe there's a way both the issues can be handled by expanding upon this idea.

**Comment by agustinhenze** [comment]
I think the following error is related to this issue but If I'm wrong please tell me and I'll open another one
```
2018-11-20 23:51:40,525 DEBUG Executing skt -vv report --reporter mail --mail-from 'CKI Gitlab <kernel-ci-list@redhat.com>' --mail-to 'skt-results-staging@redhat.com' --result /tmp/tmpkLLUl8/tmp7yEKpP --result /tmp/tmpkLLUl8/tmpb9qc_q --result /tmp/tmpkLLUl8/tmpXCNfwP --result /tmp/tmpkLLUl8/tmp2xYYfZ --mail-header 'X-Gitlab-Pipeline: https://gitlab.cee.redhat.com/cki-project/cki-pipeline/pipelines/97180' --mail-subject "[RHEL PATCH] configs: Enable checkoptions in pro
cess_configs.sh"
2018-11-20 23:51:40,647    DEBUG   No runner info found in state file, test runs will not be reported                                                                                                                                       
Traceback (most recent call last):                                                                                                                                                                                                          
  File "/home/skt/.virtualenvs/skt/bin/skt", line 11, in <module>                                                                                                                                                                           
    sys.exit(main())                                                                                                                                                                                                                        
  File "/home/skt/.virtualenvs/skt/lib/python2.7/site-packages/skt/executable.py", line 1098, in main                                                                                                                                       
    args.func(cfg)                                                                                                                                                                                                                          
  File "/home/skt/.virtualenvs/skt/lib/python2.7/site-packages/skt/executable.py", line 490, in cmd_report                                                                                                                                  
    reporter.report()                                                                                                                                                                                                                       
  File "/home/skt/.virtualenvs/skt/lib/python2.7/site-packages/skt/reporter.py", line 530, in report                                                                                                                                        
    msg.attach(MIMEText(self._get_multireport()))                                                                                                                                                                                           
  File "/home/skt/.virtualenvs/skt/lib/python2.7/site-packages/skt/reporter.py", line 404, in _get_multireport                                                                                                                              
    job_data['buildlog'] = self.__getbuildlog(kernel_arch)                                                                                                                                                                                  
  File "/home/skt/.virtualenvs/skt/lib/python2.7/site-packages/skt/reporter.py", line 214, in __getbuildlog                                                                                                                                 
    with open(self.cfg.get("buildlog"), 'r') as fileh:                                                                                                                                                                                      
IOError: [Errno 2] No such file or directory: '/cki-project/cki-pipeline/workdir/build.log'                                                                                                                                                 
```
As you can see the report won't be sent because `build.log` is missing (because the failure was on the `rh-configs` stage). I'm not sure how should we handle these cases :man_shrugging:

**Comment by agustinhenze** [comment]
Also, you can see the output here https://paste.debian.net/1052582/ (I've added the build.log by hand and re-run the same command). We are printing empty strings on the testing sections.

**Comment by veruu** [comment]
/me dug into the code for a while and the error looks related to the first point in the issue. If you and @spbnick are okay with the solutions I outlined, I can take a look at it later today (other ideas welcome!). The empty testing strings in the report are unrelated, I found out we are reporting all steps even when they didn't run while writing #395 but didn't have time to fix it yet.

**Comment by spbnick** [comment]
I'm sorry, I don't have time to look into the details of the proposed implementation right now. I would be OK with just any reasonable code change, as long as configuration generation failure is reported with exit status 1 from "skt build" (we probably shouldn't have a separate skt command for it), "skt report" can handle it, and would attach the right log to the report (it can be a part of build.log).

**Comment by veruu** [comment]
First point fixed in 7ca5d0e. We may or may not have to deal with `getrelease()` later
